hi-ml/src/health_ml/lightning_container.py

```diff
@@ -0,0 +1,171 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from typing import Any, Dict, Optional
+from pathlib import Path
+
+import param
+from azureml.core import ScriptRunConfig
+from azureml.train.hyperdrive import HyperDriveConfig
+from pytorch_lightning import LightningDataModule, LightningModule
+
+from health_ml.deep_learning_config import DatasetParams, OptimizerParams, OutputParams, TrainerParams, \
+    WorkflowParams
+from health_ml.experiment_config import ExperimentConfig
+
+
+class LightningContainer(WorkflowParams,
+                         DatasetParams,
+                         OutputParams,
+                         TrainerParams,
+                         OptimizerParams):
+    """
+    A LightningContainer contains all information to train a user-specified PyTorch Lightning model. The model that
+    should be trained is returned by the `get_model` method. The training data must be returned in the form of
+    a LightningDataModule, by the `get_data_module` method.
+    """
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self._model: Optional[LightningModule] = None
+        self._model_name = type(self).__name__
+        self.num_nodes = 1
+
+    def validate(self) -> None:
+        WorkflowParams.validate(self)
+        OptimizerParams.validate(self)
+
+    def setup(self) -> None:
+        """
+        This method is called as one of the first operations of the training/testing workflow, before any other
+        operations on the present object. At the point when called, the datasets are already available in
+        the locations given by self.local_datasets. Use this method to prepare datasets or data loaders, for example.
+        """
+        pass
+
+    def create_model(self) -> LightningModule:  # type: ignore
```

Comment:
i think somewhere in doc I saw `get_model`? Would be good to have alignment between model and data, the old code had create_model and get_data_module.

Reply:
I would propose to change this in a future PR. I have stuck with create_model for now so that config files from InnerEye will work with the new runner, to reduce the number of changes that need to be made there